msbuild/Makefile

```diff
@@ -48,6 +49,8 @@ TRANSLATED_ASSEMBLIES = $(foreach locale,$(LOCALIZATION_LANGUAGES),$(locale)/Xam
 LOCALIZATION_ASSEMBLIES = Xamarin.Localization.MSBuild
 IOS_TASK_ASSEMBLIES = Xamarin.iOS.Tasks $(LOCALIZATION_ASSEMBLIES)
 IOS_WINDOWS_TASK_ASSEMBLIES = Xamarin.iOS.Tasks.Windows
+IOS_WINDOWS_DEPENDENCIES = Xamarin.iOS.Windows.Client iSign.Core ws2_32 System.Diagnostics.Tracer BouncyCastle.Crypto System.Buffers System.Memory System.Numerics.Vectors System.Runtime.CompilerServices.Unsafe System.Security.Cryptography.ProtectedData System.Security.Cryptography.Pkcs Microsoft.Win32.Registry System.Security.AccessControl System.Security.Principal.Windows
+IOS_WINDOWS_MOBILEDEVICE_TOOLS = iMobileDevice-net bz2 getopt ideviceactivation idevicebackup idevicebackup2 idevicecrashreport idevicedate idevicedebug idevicedebugserverproxy idevicediagnostics ideviceenterrecovery ideviceimagemounter ideviceinfo ideviceinstaller idevicename idevicenotificationproxy idevicepair ideviceprovision idevicerestore idevicescreenshot idevicesyslog idevice_id imobiledevice-net-lighthouse imobiledevice ios_webkit_debug_proxy iproxy irecovery irecovery libcharset libcurl LIBEAY32 libiconv libusb-1.0 libusb0 libxml2 lzma pcre pcreposix plist plistutil plist_cmp plist_test pthreadVC3 readline SSLEAY32 usbmuxd usbmuxd vcruntime140 zip zlib1
```

Comment:
* How is this used?
* Are we bundling these ourselves if we accept the PR?
* Are we distributing this if we accept this PR?

Reply:
@dalexsoto these are the mobiledevice tools used by Hot Restart to communicate with Apple devices in Windows. We are getting them from the following public NuGet Package: https://www.nuget.org/packages/iMobileDevice-net/
The tools are already approved by Microsoft, we went though the approval process when we started the Hot Restart work for the old project system (in XamarinVS).
We are just using the binaries and the tools, we don't compile anything additionally regarding these files. The reason why we are including them this way is that we don't want and we also can't ILMerge those in just one Xamarin.iOS.Windows.Tasks.dll since they contain native code and that's not possible. 
For this reason, we just group them in specific sub folders into the Windows iOS pack so it's more organized and clean.

Reply:
Can't these live/be acquired somewhere else? These are not really dependencies of the macios repository and since they re now being part of our build and re-distributed this way then they will create unnecessary security compliance work and there are a lot of different tools/libraries listed there.

Reply:
These are dependencies of Hot Restart, which is now part of macios (not the Core library but the MSBuild targets and tasks).
They need to be part of the Windows iOS pack as any other library/dependency that we have. The only difference is that we can't ILMerge as we do with other components as Messaging in Xamarin.iOS.Tasks.
We've handled the security compliance work in XamarinVS as external components once they were approved a long time ago.
Luis can also explain better our current situation with these components, CELA and the compliance review of the last year.

Reply:
Why can't we use the nuget itself https://www.nuget.org/packages/iMobileDevice-net/ instead of re-packaging and re-distribute everything?

Reply:
That iMobileDevice-net NuGet package is currently a PackageReference of the Xamarin.iOS.Windows.Client NuGet package that we reference here: https://github.com/xamarin/xamarin-macios/blob/net6-hr/msbuild/Xamarin.iOS.Tasks.Windows/Xamarin.iOS.Tasks.Windows.csproj#L15

The point is that we need to bundle the files in the net6 pack for Windows, that's all we are doing. We are not referencing them directly in any project but just copying them to the net6 pack for Windows so when the pack is loaded at runtime on a user machine it can satisfy all the dependencies. We are doing the same for any other dependency, ensuring we copy them to the net6 resulting pack.

AFAIK no net6 pack or SDK is resolving Package References at runtime, please correct me If I'm wrong.